# Housing Market

### Introduction:

This time we will create our own dataset with fictional numbers to describe a house market. As we are going to create random data don't try to reason of the numbers.

### Step 1. Import the necessary libraries

In [1]:
import pandas as pd
import numpy as np

### Step 2. Create 3 differents Series, each of length 100, as follows: 
1. The first a random number from 1 to 4 
2. The second a random number from 1 to 3
3. The third a random number from 10,000 to 30,000

In [8]:
serie_1 = pd.Series(np.random.randint(1, 5, 100))
serie_2 = pd.Series(np.random.randint(1, 4, 100))
serie_3 = pd.Series(np.random.randint(10000, 30001, 100))

### Step 3. Let's create a DataFrame by joinning the Series by column

In [17]:
df_series = pd.DataFrame([serie_1, serie_2, serie_3]).T

In [18]:
df_series

0  1      2
0   4  1  20933
1   3  3  17974
2   1  2  26925
3   3  2  19266
4   2  1  17699
.. .. ..    ...
95  2  2  26470
96  1  3  29562
97  2  2  17197
98  2  3  12018
99  1  1  14529

[100 rows x 3 columns]

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [19]:
df_series.columns = ['bedrs', 'bathrs', 'price_sqr_meter']

In [20]:
df_series

bedrs  bathrs  price_sqr_meter
0       4       1            20933
1       3       3            17974
2       1       2            26925
3       3       2            19266
4       2       1            17699
..    ...     ...              ...
95      2       2            26470
96      1       3            29562
97      2       2            17197
98      2       3            12018
99      1       1            14529

[100 rows x 3 columns]

### Step 5. Create a column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [21]:
pd.DataFrame(pd.concat([serie_1, serie_2, serie_3]).reset_index(drop=True), columns = ['bigcolumn'])

bigcolumn
0            4
1            3
2            1
3            3
4            2
..         ...
295      26470
296      29562
297      17197
298      12018
299      14529

[300 rows x 1 columns]

### Step 6. Oops, it seems it is going only until index 99. Is it true?

In [25]:
df_series_99 = pd.DataFrame(pd.concat([serie_1, serie_2, serie_3]))
pd.set_option('display.max_rows', 99)

In [26]:
df_series_99

0
0       4
1       3
2       1
3       3
4       2
..    ...
95  26470
96  29562
97  17197
98  12018
99  14529

[300 rows x 1 columns]

### Step 7. Reindex the DataFrame so it goes from 0 to 299

In [27]:
df_series_99.reset_index(drop = True)

0
0        4
1        3
2        1
3        3
4        2
..     ...
295  26470
296  29562
297  17197
298  12018
299  14529

[300 rows x 1 columns]